lib/src/main/java/io/ably/lib/objects/LiveCounter.java

```diff
@@ -0,0 +1,46 @@
+package io.ably.lib.objects;
+
+import io.ably.lib.types.Callback;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Contract;
+
+/**
+ * The LiveCounter interface provides methods to interact with a live counter.
+ * It allows incrementing, decrementing, and retrieving the current value of the counter,
+ * both synchronously and asynchronously.
+ */
+public interface LiveCounter {
+
+    /**
+     * Increments the value of the counter by 1.
+     */
+    void increment();
+
+    /**
+     * Increments the value of the counter by 1 asynchronously.
+     *
+     * @param callback the callback to be invoked upon completion of the operation.
+     */
+    void incrementAsync(@NotNull Callback<Void> callback);
```

Comment:
I guess async methods can have nullable callbacks ... but doesn't serve a purpose when you don't handle success/ error of given operation. 
So better to make sure every operation result is properly handled.

Reply:
I agree, `foo.bar(null)` doesn't look good! on other hand, sometimes it's handy to fire and forget, in that case we can overload and add `void incrementAsync()`. But let's wait until we have some real-world use-cases